test/xds/xds_client_ignore_resource_deletion_test.go

```diff
@@ -0,0 +1,473 @@
+			if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+				return
+			}
+		}
+	}()
+	go func() {
+		defer wg.Done()
+		for ; ctx.Err() == nil; <-time.After(10 * time.Millisecond) {
+			if _, err := client.UnaryCall(ctx, &testpb.SimpleRequest{}); err != nil {
+				return
+			}
+		}
+	}()
+
+	wg.Wait()
+	if ctx.Err() != nil {
+		t.Fatal("Context expired before RPCs failed.")
+	}
+}
+
+// This helper creates a management server for the test.
+func startManagementServer(t *testing.T) *e2e.ManagementServer {
+	t.Helper()
+	mgmtServer, err := e2e.StartManagementServer(e2e.ManagementServerOptions{})
+	if err != nil {
+		t.Fatalf("Failed to start management server: %v", err)
+	}
+	t.Cleanup(mgmtServer.Stop)
+	return mgmtServer
+}
+
+// This helper generates a custom bootstrap config for the test.
+func generateBootstrapContents(t *testing.T, serverURI string, ignoreResourceDeletion bool) []byte {
+	t.Helper()
+	bootstrapContents, err := bootstrap.Contents(bootstrap.Options{
+		NodeID:                             nodeID,
+		ServerURI:                          serverURI,
+		ServerListenerResourceNameTemplate: e2e.ServerListenerResourceNameTemplate,
+		IgnoreResourceDeletion:             ignoreResourceDeletion,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	return bootstrapContents
+}
+
+// This helper creates in a bootstrapConfig and .
+func xdsResolverBuilder(t *testing.T, bs []byte) resolver.Builder {
+	t.Helper()
+	resolverBuilder := internal.NewXDSResolverWithConfigForTesting.(func([]byte) (resolver.Builder, error))
+	xdsR, err := resolverBuilder(bs)
+	if err != nil {
+		t.Fatalf("Creating xDS resolver for testing: %v", err)
+	}
+	return xdsR
+}
+
+// This helper creates an xDS-enabled gRPC server using the listener and the
+// bootstrap config passed. This helper then registers the server to testService
+// and returns a func to accept requests.
+func setupGRPCServerWithModeChangeChannel(t *testing.T, bootstrapContents []byte, lis net.Listener) (chan connectivity.ServingMode, func()) {
+	t.Helper()
+	updateCh := make(chan connectivity.ServingMode, 1)
+
+	// Create a server option to get notified about serving mode changes.
+	modeChangeOpt := xds.ServingModeCallback(func(addr net.Addr, args xds.ServingModeChangeArgs) {
+		t.Logf("serving mode for listener %q changed to %q, err: %v", addr.String(), args.Mode, args.Err)
+		updateCh <- args.Mode
+	})
+
+	server := xds.NewGRPCServer(grpc.Creds(insecure.NewCredentials()), modeChangeOpt, xds.BootstrapContentsForTesting(bootstrapContents))
+	t.Cleanup(server.Stop)
+	testpb.RegisterTestServiceServer(server, &testService{})
+
+	return updateCh, func() {
+		if err := server.Serve(lis); err != nil {
+			t.Errorf("Serve() failed: %v", err)
+		}
+	}
+}
+
+// This helper creates a listener resource with using a custom listener created.
+// The test uses this listener to serve the gRPC server that is set up.
+func resourceWithListenerForGRPCServer(t *testing.T) (e2e.UpdateOptions, net.Listener) {
```

Comment:
I feel like it would be better if we make this helper to only create the local listener and return its host and port.

So, something like `createLocalListenerForServer(t *testing.T) (lis net.Listener, host string, port uint32)`.

Then, the test would do:
```golang
	lis, host, port := resourceWithListenerForGRPCServer(t)
	updateCh  := setupGRPCServerWithModeChangeChannel(t, bs, lis)

	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
	defer cancel()
	if err := mgmtServer.Update(ctx, e2e.UpdateOptions{
		NodeID:    nodeID,
		Listeners: []*listenerpb.Listener{e2e.DefaultServerListener(host, port, e2e.SecurityLevelNone)},
	}); err != nil {
		t.Fatal(err)
	}

```